### ISSUE TITLE
Browser instance on a network with proxy

### ISSUE THREAD

**Issue by cassionandi** [cobrateam/splinter#120]
Hi.

I'm trying to run Splinter in a Windows 7 machine, in a local network that needs a proxy.

When a new instance of the Browser() is created in the python shell, a firefox window pop-ups with the profile different than the one that I use. So, this new instance and profile don't have none of the configs for proxy.

This is the message generated by the "browser = Browser()" code:

Traceback (most recent call last):
  File "C:\Users\cassio.nandi\Desktop\splinter\exemplo.py", line 3, in <module>
    browser = Browser()
  File "C:\Python27\lib\site-packages\splinter\browser.py", line 46, in Browser
    return driver(_args, *_kwargs)
  File "C:\Python27\lib\site-packages\splinter\driver\webdriver\firefox.py", line 23, in **init**
    self.driver = Firefox(firefox_profile)
  File "C:\Python27\lib\site-packages\selenium\webdriver\firefox\webdriver.py", line 47, in **init**
    desired_capabilities=DesiredCapabilities.FIREFOX)
  File "C:\Python27\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 61, in **init**
    self.start_session(desired_capabilities, browser_profile)
  File "C:\Python27\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 98, in start_session
    'desiredCapabilities': desired_capabilities,
  File "C:\Python27\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 144, in execute
    self.error_handler.check_response(response)
  File "C:\Python27\lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 100, in check_response
    raise exception_class(value)
WebDriverException: Message: '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">\n<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">\n<TITLE>ERRO: A URL solicitada n&atilde;o pode ser recuperada</TITLE>\n<STYLE type="text/css"></STYLE>\n</HEAD><BODY>\n<H1>ERRO</H1>\n<H2>A URL solicitada n&atilde;o pode ser recuperada</H2>\n<HR noshade size="1px">\n<P>\nNa tentativa de recuperar a URL:\n<A HREF="http://127.0.0.1:51372/hub/session">http://127.0.0.1:51372/hub/session</A>\n<P>\nO seguinte erro foi encontrado:\n<UL>\n<LI>\n<STRONG>\nProibido o Acesso.\n</STRONG>\n<P>\nO controle de acessos impediu sua requisi&ccedil;&atilde;o.\nCaso voc&ecirc; n&atilde;o concorde com isso, por favor, \ncontate seu provedor de servi&ccedil;os, ou o administrador de sistemas.\n</UL>\n</P>\n\n\n<BR clear="all">\n<HR noshade size="1px">\n<ADDRESS>\nGenerated Tue, 08 Nov 2011 18:27:08 GMT by proxy (squid/2.6.STABLE21)\n</ADDRESS>\n</BODY></HTML>\n'

**Comment by yesbox** [comment]
This issue blocks the usage of Splinter in environments that require the use of an HTTP proxy. This appears to be a long-standing issue, is there a known workaround?

This script is used in the following scenarios on a headless CentOS 7 server with XVFB, Firefox 38 and Splinter 0.7.2. tcpdump running to to analyze its behavior:
```
from splinter.browser import Browser

fx_config = {
#    'network.proxy.type': 1,
#    'network.proxy.http': 'internal.proxy.local',
#    'network.proxy.http_port': 8080,
    'browser.startup.homepage': 'http://www.example-site-1.net/',
}

browser = Browser('firefox', profile_preferences=fx_config)
browser.visit('http://www.example-site-2.net/')
```

### Baseline:
Without an http_proxy env variable set the browser can be seen resolving site 1 and site 2 DNS names and attempting to open HTTP connections to both of them (along with some Mozilla services), all of which will fail due to the restrictive network environment, eventually raising "socket.error: [Errno 101] Network is unreachable".

### http_proxy env variable set:
When setting the http_proxy env variable and running the script the HTTP proxy DNS name is looked up the browser opens a connection to the server on the proxy port. No connections to site 1 is seen. After this, the site 2 DNS name is resolved and the browser attempts to connect to directly to it. This too fails.

### Firefox proxy settings applied:
When uncommenting the network.proxy settings in the script, settings the Firefox HTTP proxy preferences as has previously been suggested in this thread, the behavior appears identical to the baseline, regardless of whether the http_proxy env variable is set or not. I'm not certain if these settings are correct, but they do match the changes found under about:config when I setup and successfully use Firefox with the same proxy server via the UI on my desktop machine.

To summarize, it appears the browser.visit method does not respect the http_proxy env variable like other connections Firefox opens when used with Splinter. In addition it appears that setting network.proxy settings via Splinter breaks HTTP proxy browsing further, attempting to use direct connections both with and without the browser.visit call.

**Comment by andrewsmedina** [comment]
@yesbox I have pushed a commit that fix this problem. Can you test the master version?

**Comment by yesbox** [comment]
Running master commit e47f044a398271bf91374999ec23d726d3f06075 against an HTTP proxy with the http_proxy and https_proxy variable set now works.

Using the profile_preferences to configure HTTP proxy in the same manner as in the previous example still breaks it.

Both http_proxy and https_proxy must be set, either used alone will not give any response. I did not test these env variables separately previously but always used both when testing.

Thanks!

**Comment by andrewsmedina** [comment]
after the https://github.com/cobrateam/splinter/commit/2e5135460610149a1082b2aaca61fffc10beae40 in `0.8` the drivers based on webdriver will `raise NotImplementedError` in `status_code`
